pages/favs.jsx

```diff
@@ -1,20 +1,69 @@
 import React from 'react';
+import { browserHistory } from 'react-router';
 import { getFavs } from '../js/favs-manager';
 import ProjectList from '../components/project-list/project-list.jsx';
 import HintMessage from '../components/hint-message/hint-message.jsx';
 
+import Service from '../js/service.js';
+import config from '../config/config.js';
+import env from "../config/env.generated.json";
+
 export default React.createClass({
   getInitialState() {
     return {
       favs: null
     };
   },
+  verifyLoggedInStatus() {
+    Service.userstatus()
+      .then(response => this.updateLoggedInInfo(response.loggedin, response.username))
+      .catch(reason => {
+        this.updateLoggedInInfo(false);
+        console.error(reason);
+      });
+  },
+  updateLoggedInInfo(status, username = ``) {
+    config.userLoggedInStatus.set(status);
+    config.username.set(username);
+    this.forceUpdate();
+  },
+  logOutUser() {
+    Service.logout()
+      .then(() => {
+        // we have successfully logged user out
+        this.updateLoggedInInfo(false);
+      })
+      .catch(reason => {
+        console.error(reason);
+      });
+  },
   componentDidMount() {
+    let query = this.props.router.location.query;
+    
+    // we don't need the "loggedin" param sent back from Pulse API to be presented at all times
+    delete query.loggedin;
+
+    browserHistory.push({
```

Comment:
you want a `replace` here, rather than a `push`, so that the browser "forgets" there was ever a `loggedin` query argument to begin with.